inc/compat/plugins/compat-plugin-mailpoet.php

```diff
@@ -0,0 +1,137 @@
+<?php
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Compatibility with plugin: MailPoet integration.
+ */
+class FluidCheckout_MailPoet extends FluidCheckout {
+
+    /**
+     * __construct function.
+     */
+    public function __construct( ) {
+        add_action( 'init', [ $this, 'hooks' ] );
+    }
+
+    /**
+     * Initialize hooks.
+     */
+    public function hooks() {
+        // Bail when no mail poet class
+        if ( ! class_exists( MailPoet\Config\HooksWooCommerce::class ) ) {
+            return;
+        }
+
+        $this->remove_class_action( 'woocommerce_checkout_before_terms_and_conditions', 'MailPoet\Config\HooksWooCommerce',  'extendWooCommerceCheckoutForm' );
+        add_action( 'fc_checkout_contact_after_fields', [ $this, 'extend_woocommerce_checkout_form' ], 10 );
+    }
+
+    /**
+     * Extend WooCommerce checkout form for checkbox
+     *
+     * @return html
```

Comment:
@alamgircsebd as an "action" hook, this function does not return any value, only `void` when bailing. Please remove this.